### DISCUSSION TITLE
Definition of the available map restore types

### DISCUSSION THREAD

**Discussion by pidator** [rand256/valetudo#340] [Q&A] [answered]
There are four types of map restoring available in settings -> persistent data configuration:
- Default
- Service Restart
- Cleaning Sequence
- Backup Recovery

What is the difference between these four types and what method should be used?
Is it depending on the vacuum model or the base firmware? What type of map restoring is used when loading an existing map via home tab?
Does this setting influences the general map restoring on the robot or only the auto restore function when returning to the dock?

Thanks.

**Comment by rand256** [comment, marked as answer]
These types define map loading method valetudo would use on the device in all the corresponding features: in Auto Map Restore and both manual map loading on the Home tab of WebUI or via "load_map" MQTT command. All the methods replaces some map files and then make firmware re-read them, which can be done differently depending on the device.

- **Default** is in fact the auto-select of the "best" possible type available.
- **Service Restart** simply kills some manufacturer's services running on the device and waits them to restart, so they could reapply the changed maps. This is the only option available for gen1 devices, and the slowest one to complete.
- **Cleaning Sequence** method utilizes the fact (maybe a bug?) that if you have persistent maps enabled and quickly call _start cleaning_ + _stop cleaning_, then you get firmware re-read maps from files. The downside of this method is the device will try to announce the beginning and ending of the cleaning. The engine in most cases won't start rotating though. This is the second option available for gen2 devices before 2008 firmware appeared.
- **Backup Recovery** method uses the special command first appeared in gen3 devices and then ported to gen2 with 2008+ firmware. It just... restores the map - quick and silent. Although there were some claims that it may lose forbidden zones or something during processing, I guess that was rather a glitch but not a consistent behavior for this method.

So, in most cases just leaving this option on "Default" should suffice for everyone.
